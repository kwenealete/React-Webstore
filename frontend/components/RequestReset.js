import Form from './styles/Form';
import useForm from '../lib/useForm';
import gql from 'graphql-tag';
import { useMutation } from '@apollo/client';
import Error from './ErrorMessage';

const REQUEST_RESET_MUTATION = gql`
    mutation REQUEST_RESET_MUTATION($email: String!,) {
        sendUserPasswordResetLink (email: $email) {
            code
            message
            
        }
    }
`;

export default function RequestReset() {
    const { inputs, handleChange, resetForm } = useForm({
        email: '',
    });

    const [signup, { data, loading, error }] = useMutation(REQUEST_RESET_MUTATION, {
        variables: inputs,
       
    });

    async function handleSubmit(e) {
        e.preventDefault();
        console.log(inputs);
        const res = await signup().catch(console.error);
        console.log(res);
        resetForm();
        
    }
   
    return (
        <Form method="POST" onSubmit={handleSubmit}>
            <h2>Forgotten your password? Request a new password</h2>
            <Error error={error} />
            <fieldset>
                {data?.sendUserPasswordResetLink === null && (<p> Success! Check email for link </p>)}
                <label htmlFor="email">
                    Email
                    <input 
                        type="email"
                        name="email"
                        placeholder="Enter your email address"
                        autoComplete="email"
                        value={inputs.email}
                        onChange={handleChange}

                    />
                </label>
                
                <button type="submit">Request Reset!</button>
            </fieldset>
        </Form>
    );
    
}

export { REQUEST_RESET_MUTATION };