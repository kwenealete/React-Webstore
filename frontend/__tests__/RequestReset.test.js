import { render, screen } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import userEvent from '@testing-library/user-event';
import RequestReset, { REQUEST_RESET_MUTATION } from '../components/RequestReset';

const email = 'alete@yahoo.com';

const mocks = [
    {
        request: {
            query: REQUEST_RESET_MUTATION,
            variables: {email},
        },
        result: {
            data: { sendUserPasswordResetLink: null }
        },
    },
];

describe('<RequestReset />', () => {
    it('Renders and matches snapshot', () => {
        const { container, debug } = render (
            <MockedProvider>
                <RequestReset />
            </MockedProvider>
        );
        expect(container).toMatchSnapshot();
    });

    it('Calls the mutation when requested', async() => {
        const{ container, debug } = render(
            <MockedProvider mocks={mocks} >
                <RequestReset />
            </MockedProvider>
        );
         // type into the email box
        userEvent.type(screen.getByPlaceholderText(/email/i), email);
        // click submit
        userEvent.click(screen.getByText(/Request Reset!/));
        const success = await screen.findByText(/Success/i);
        expect(success).toBeInTheDocument();
    });
})