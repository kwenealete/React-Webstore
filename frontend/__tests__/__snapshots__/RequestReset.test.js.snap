// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset /> Renders and matches snapshot 1`] = `
<div>
  <form
    class="Form-sc-1xszr8q-0 hVEEdl"
    method="POST"
  >
    <h2>
      Forgotten your password? Request a new password
    </h2>
    <fieldset>
      <label
        for="email"
      >
        Email
        <input
          autocomplete="email"
          name="email"
          placeholder="Enter your email address"
          type="email"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Request Reset!
      </button>
    </fieldset>
  </form>
</div>
`;
